llvm/include/llvm/Option/OptTable.h

```diff
@@ -117,13 +121,13 @@ class OptTable {
 private:
   // A unified string table for these options. Individual strings are stored as
   // null terminated C-strings at offsets within this table.
-  const char *StrTable;
+  const StringTable *StrTable;
```

Comment:
Can this be a const reference instead? Is there a reason you chose to make this a pointer?

Reply:
Using a const reference makes the type unassignable (without UB) sadly, because references can't be re-bound.

Technically, this type could probably be unassignable, but I always prefer regular types that don't have complex constraints like that and so I much prefer pointers rather than references as members.

That said, if there is a strong preference, I can try switching to a reference and see if I can get it all to compile.

Reply:
Got it, and I do seem to recall we sometimes use OptTable as a value type, even if that is somewhat questionable.